src/modules/powerrename/lib/Settings.cpp

```diff
@@ -470,12 +359,113 @@ void CRenameMRU::_FreeMRUList()
     }
 }
 
+CSettings::CSettings()
+{
+    std::wstring result = PTSettingsHelper::get_module_save_folder_location(L"PowerRename");
+    jsonFilePath = result + L"\\" + std::wstring(c_powerRenameDataFilePath);
+}
+
+void CSettings::LoadPowerRenameData()
+{
+    if (!std::filesystem::exists(jsonFilePath))
+    {
+        MigrateSettingsFromRegistry();
+
+        SavePowerRenameData();
+    }
+    else
+    {
+        ParseJsonSettings();
+    }
+}
+
+void CSettings::SavePowerRenameData() const
+{
+    json::JsonObject jsonData;
+
+    jsonData.SetNamedValue(c_enabled,                 json::value(settings.enabled));
+    jsonData.SetNamedValue(c_showIconOnMenu,          json::value(settings.showIconOnMenu));
+    jsonData.SetNamedValue(c_extendedContextMenuOnly, json::value(settings.extendedContextMenuOnly));
+    jsonData.SetNamedValue(c_persistState,            json::value(settings.persistState));
+    jsonData.SetNamedValue(c_mruEnabled,              json::value(settings.MRUEnabled));
+    jsonData.SetNamedValue(c_maxMRUSize,              json::value(settings.maxMRUSize));
+    jsonData.SetNamedValue(c_flags,                   json::value(settings.flags));
+    jsonData.SetNamedValue(c_searchText,              json::value(settings.searchText));
+    jsonData.SetNamedValue(c_replaceText,             json::value(settings.replaceText));
+
+    json::to_file(jsonFilePath, jsonData);
+}
+
+void CSettings::MigrateSettingsFromRegistry()
+{
+    settings.enabled                 = GetRegBoolean(c_enabled, true);
+    settings.showIconOnMenu          = GetRegBoolean(c_showIconOnMenu, true);
+    settings.extendedContextMenuOnly = GetRegBoolean(c_extendedContextMenuOnly, false); // Disabled by default.
+    settings.persistState            = GetRegBoolean(c_persistState, true);
+    settings.MRUEnabled              = GetRegBoolean(c_mruEnabled, true);
+    settings.maxMRUSize              = GetRegNumber(c_maxMRUSize, 10);
+    settings.flags                   = GetRegNumber(c_flags, 0);
+    settings.searchText              = GetRegString(c_searchText);
+    settings.replaceText             = GetRegString(c_replaceText);
+}
+
+void CSettings::ParseJsonSettings()
+{
+    auto json = json::from_file(jsonFilePath);
+    if (json)
+    {
+        const json::JsonObject& jsonSettings = json.value();
+        if (json::has(jsonSettings, c_enabled, json::JsonValueType::Boolean))
+        {
+            settings.enabled = jsonSettings.GetNamedBoolean(c_enabled);
+        }
+        if (json::has(jsonSettings, c_showIconOnMenu, json::JsonValueType::Boolean))
+        {
+            settings.showIconOnMenu = jsonSettings.GetNamedBoolean(c_showIconOnMenu);

```

Comment:
`GetNamedBoolean` and the other json get methods throw exceptions, we need to catch them.

Reply:
You are right, before I was using the ones with default parameter which don't throw exception, so I forgot to add it now. Will change.

Reply:
But `GetNamedBoolean` is still a method can could potentially throw an exception, so we have to catch it even if we reduce the chance to happen.

Reply:
Added try-catch block.